File: common/discord/types/payloads/gateway/events/message-reaction-remove.go
package events

import "github.com/discordnova/nova/common/discord/types/structures"

// GatewayEventMessageReactionRemove is sent when a user removes a reaction from a message.
type GatewayEventMessageReactionRemove struct {
	// the id of the user
	UserID string `json:"user_id"`
	// the id of the channel
	ChannelID string `json:"channel_id"`
	// the id of the message
	MessageID string `json:"message_id"`
	// the id of the guild
	GuildID string `json:"guild_id,omitempty"`
	// the emoji used to react
	Emoji structures.Emoji `json:"emoji"`
}

File: common/rust/src/types/dispatch/message_reaction_remove.rs
use crate::types::emoji::Emoji;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct MessageReactionRemove {
    user_id: String,
    channel_id: String,
    message_id: String,
    guild_id: Option<String>,
    emoji: Emoji,
}